### PR TITLE
94 doc release procedure

### LINKED ISSUE #94: Document release process and include actual release notes in Changelog file
The release process doesn't appear to be documented very well (e.g. with a checklist). The past few release PRs did not have without any description (e.g. #68, #73, #76), and the changelog page at https://tudelftgeodesy.github.io/sarxarray/CHANGELOG/ simply points to https://github.com/TUDelftGeodesy/sarxarray/releases even though it states "All notable changes to this project will be documented in this file".

https://github.com/TUDelftGeodesy/sarxarray/blob/3a11cd07450045b74a8418135d2aa949f32a87fa/docs/CHANGELOG.md?plain=1#L1-L6

Recommendations:
1. Actually include release notes in CHANGELOG.md (refer to https://www.pyopensci.org/python-package-guide/documentation/repository-files/changelog-file.html on formatting)
2. Create a checklist, possibly as an Issue/Pull Request template, so that maintainers know how to create a release. Include steps such as:
    - [ ] Update CHANGELOG.md file
    - [ ] Update version in `pyproject.toml`
    - [ ] Create release/tag at https://github.com/TUDelftGeodesy/sarxarray/releases

Note that these are only suggestions, and you're welcome to implement some or all of the above 🙂 There are also other 'advanced' methods of e.g. updating the version based on git tags (https://www.pyopensci.org/python-package-guide/package-structure-code/python-package-versions.html#tool-3-setuptools-scm-versioning-using-git-tags), but up to you on what you're comfortable with implementing.

Part of JOSS review at https://github.com/openjournals/joss-reviews/issues/10492

### PR DESCRIPTION
**Pull request by rogerkuou**
fix #94 

- Update Changelog
- Add a GitHub issue template for release checklist
- Add an agent skill for generating changelog infor
- Update JOSS paper AI usage disclosure section.

### CODE REVIEW

**Comment by rogerkuou** [comment]
Hi @SarahAlidoost , when you have time, can you help me review this PR?

This PR is to fix a comment we got from the JOSS review, where the reviewer asked us to improve the Change Log, create templates to standardize the release process.

While working on these suggestions, I think I can go one step further and created an agent skill to generate draft CHANGELOG info. This is the part I specifically want to ask for your input.

You can run in GitHub Copilot `/release-changelog v1.2.4` to locally test the agent skill for the next release. This is also documented in the GitHub issue template.

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+name: Release Checklist
+about: Create a checklist for making a new release.
+title: Release checklist vX.Y.Z
+labels: ''
+assignees: ''
+type: Task
+
+---
+
+I will make a new release by preforming the following procedures:
```
Comment (SarahAlidoost) [inline .github/ISSUE_TEMPLATE/release-checklist.md]:
```suggestion
I will make a new release by preforming the following steps:
```

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+- [ ] Step1: Update version in pyproject.toml
+- [ ] Step2: Update `docs/CHANGELOG.md` with new change information
+- [ ] Step3: Create release/tag at https://github.com/TUDelftGeodesy/sarxarray/releases
+
+Hints:
+- Step 1 and 2 can be done semi-automatically by calling agent skill in [release-changelog](.github/skills/release-changelog). However the changes will always be reviewed and commited by human.
+- The agent skill can be executed in GenAI prompt, e.g. GitHub Copilot Chat, by `/release-changelog vX.Y.Z` command, where `vX.Y.Z` is the new version label.
+- Alternatively, one can run the helper script `bash .github/skills/release-changelog/scripts/generate_release_section.sh vX.Y.Z` to get bullets for the new release section.
```
Comment (SarahAlidoost) [inline .github/ISSUE_TEMPLATE/release-checklist.md]: These are documentation on how the agent skill can be called. I suggest moving these two lines to the `CONTRIBUTING.md` under the section **"Making a release"**.

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+---
+
+I will make a new release by preforming the following procedures:
+
+- [ ] Step1: Update version in pyproject.toml
+- [ ] Step2: Update `docs/CHANGELOG.md` with new change information
+- [ ] Step3: Create release/tag at https://github.com/TUDelftGeodesy/sarxarray/releases
+
+Hints:
+- Step 1 and 2 can be done semi-automatically by calling agent skill in [release-changelog](.github/skills/release-changelog). However the changes will always be reviewed and commited by human.
```
Comment (SarahAlidoost) [inline .github/ISSUE_TEMPLATE/release-checklist.md]: Is there a documentation on how to call the agent if so, please add a link here.

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+type: Task
+
+---
+
+I will make a new release by preforming the following procedures:
+
+- [ ] Step1: Update version in pyproject.toml
+- [ ] Step2: Update `docs/CHANGELOG.md` with new change information
+- [ ] Step3: Create release/tag at https://github.com/TUDelftGeodesy/sarxarray/releases
+
```
Comment (SarahAlidoost) [inline .github/ISSUE_TEMPLATE/release-checklist.md:16]: In addition to this issue template, release guide should be added to the `CONTRIBUTING.md` under a section "**Making a release**", see [example](https://github.com/NLeSC/python-template/blob/main/CONTRIBUTING.md#you-want-to-make-a-new-release-of-the-code-base).

File: .github/skills/release-changelog/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+user-invocable: true
+---
+
+# Release Changelog Skill
+
+Use this skill to generate a changelog section and insert it into docs/CHANGELOG.md before creating a GitHub release.
+
+## When To Use
+
+- You just created a new release tag and want changelog content.
```
Comment (SarahAlidoost) [inline .github/skills/release-changelog/SKILL.md]: Should the changelog be created/update before the release tag?

File: .github/skills/release-changelog/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+## Commands
+
+- Generate using auto-detected previous tag:
+  - bash .github/skills/release-changelog/scripts/generate_release_section.sh vX.Y.Z
+- Generate with explicit range:
+  - bash .github/skills/release-changelog/scripts/generate_release_section.sh vX.Y.Z vX.Y.(Z-1)
+
+Notes:
+- This supports changelog-first workflows where docs are built from release-triggered actions.
```
Comment (SarahAlidoost) [inline .github/skills/release-changelog/SKILL.md]: This sentence is not clear.

File: docs/CHANGELOG.md [hunk 7]
```diff
@@ -4,3 +4,113 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
 See [releases](https://github.com/TUDelftGeodesy/sarxarray/releases) in the sarxarray repository for more details.
+
+The information in this file can be generated automatically by LLM using the [release-changelog skill](.github/skills/release-changelog/SKILL.md). However, the generated content has always been reviewed and if necessary, edited by a human.
```
Comment (SarahAlidoost) [inline docs/CHANGELOG.md]:
```suggestion
The information in this file can be generated automatically by LLM using the [release-changelog skill](../.github/skills/release-changelog/SKILL.md). However, the generated content has always been reviewed and if necessary, edited by a human.
```

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 3 above)
Comment (SarahAlidoost) [inline .github/ISSUE_TEMPLATE/release-checklist.md]:
```suggestion
- Step 1 and 2 can be done semi-automatically by calling agent skill, see [release-changelog](.github/skills/release-changelog). However the changes will always be reviewed and commited by human.
```

File: paper/paper.md [hunk 8]
```diff
@@ -84,6 +84,8 @@ We would also like to thank Dr. Francesco Nattino and Dr. Meiert Willem Grootes
 
 In writing the software documentation and the JOSS paper, GPT-5 was used for language improvements for grammar correction and style edits.
 
+Specifically, in the documentation process, an agent skill (under path `.github/skills/release-changelog`) is used to generate changelog entries from git tags and commit history. The generated content was always reviewed and edited by a human before being committed to the repository.
```
Comment (SarahAlidoost) [inline paper/paper.md]:
```suggestion
Specifically, in the documentation process, an agent skill (under path `../.github/skills/release-changelog`) is used to generate changelog entries from git tags and commit history. The generated content was always reviewed and edited by a human before being committed to the repository.
```

**Review by SarahAlidoost — APPROVED** [review]
@rogerkuou looks good! 👍 Nice to see how SKILLS can be created for coding agents and how tasks can be done by agents. I haven't run the agent though, but I could run the script. I left some comments.

**Comment by SarahAlidoost** [comment]
> @rogerkuou looks good! 👍 Nice to see how SKILLS can be created for coding agents and how tasks can be done by agents. I haven't run the agent though, but I could run the script. I left some comments.

To follow up, I could run the agent successfully. I ran `/release-changelog v1.2.4` and got the expected changes to CHANGELOG and pyproject.toml:

```bash
git diff docs/CHANGELOG.md
```
```bash
+## Unreleased
+
+## [v1.2.4] - 2026-07-01
+
+Compared to: v1.2.3..HEAD (pre-tag draft)
+
+### Added
+- Add an issue template
+- Add agent skills for generating CHANGELOG
+- Add read_metadata to_binary and crop to API docs
+- Add output file test for to_binary
+- Fix linting, add tests
+- Add Simon as an author
+- Add a data downloading cell in the example notebook
+- Add more sections
+- Add complex coherence computation
+- Add SURF grant numbers
+- Add GitHub Action for paper
+- Add draft paper
+
+### Changed
+- Update issue template
+- Update AI usage disclosure section in JOSS paper
+- Update CHANGELOG
+- Update citation file
+- Remove list ensurance since this is already done in read_metadat
a
+- Update installation guide in docs
+- Update README with extra dependencies and zsh safe format
+- Update paper
+- Update software name in README to SARXarray
+- Update workflow
+
+### Fixed
+- Fix DOI links in JOSS paper
+- Fix typo
+
+### Docs
+- Update docs/api_reference.md
+- Add missing tickmark in docstring
+- Fix docstring
+
+### CI
+- Split filewriting test in dataset and dataarray
+- Test more edge cases in io
+- Remove da_crop reference in ValueError test
+- Add ValueError test
+- Add crop test, allow for non-integer crops
+
```

File: .github/skills/release-changelog/SKILL.md (hunk 5 above)
Comment (rogerkuou) [reply .github/skills/release-changelog/SKILL.md]: Indeed! I updated to **You are preparing a new release and want to generate a new section in the changelog.**

File: .github/skills/release-changelog/SKILL.md (hunk 6 above)
Comment (rogerkuou) [reply .github/skills/release-changelog/SKILL.md]: I removed this sentence. This was coming from a try, when I wanted enforce the agent to understand that the changlog information needs to be updated before a release is made. Now I found it's not necessary.

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 4 above)
Comment (rogerkuou) [reply .github/ISSUE_TEMPLATE/release-checklist.md:16]: Added this section to CONTRIBUTING.md

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 3 above)
Comment (rogerkuou) [reply .github/ISSUE_TEMPLATE/release-checklist.md]: Pointing to CONTRIBUTING.md now

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 2 above)
Comment (rogerkuou) [reply .github/ISSUE_TEMPLATE/release-checklist.md]: Moved this to COONTRIBUTING.md

**Comment by rogerkuou** [comment]
Thanks @SarahAlidoost . I have adapted all your comments. Merging now.
